lib/api.js

```diff
@@ -128,13 +139,7 @@ var api = {
     reqOptions.files = options.files;
 
     if (options.auth === true) {
-      // Runtime fetch of Auth singleton to prevent circular module dependencies
-      var auth = require('./auth');
-
-      return auth.getAccessToken(refreshToken, commandScopes).then(function(result) {
-        reqOptions.headers = {
-          authorization: 'Bearer ' + result.access_token
-        };
+      return api.addAccessTokenToHeader(reqOptions).then(function() {
```

Comment:
I know this is technically modifying the reqOptions because it's being passed by reference, but I think it'd be more readable if you used the output from the promise, e.g. (with perhaps a better name for `result`)
```js
return api.addAccessTokenToHeader(reqOptions).then(function(result) {
  return _request(result);
});
```